waterbutler/providers/rushfiles/provider.py

```diff
@@ -123,7 +123,22 @@ async def download(self,  # type: ignore
                        revision: str=None,
                        range: Tuple[int, int]=None,
                        **kwargs) -> streams.BaseStream:
-        raise NotImplementedError
+        if path.identifier is None:
+            raise exceptions.DownloadError('"{}" not found'.format(str(path)), code=404)
+
+        if path.is_dir:
+            raise exceptions.DownloadError('Path must be a file', code=404)
+
+        metadata = await self.metadata(path)
+            
+        resp = await self.make_request(
+            'GET',
+            self._build_filecache_url(str(self.share['id']), 'virtualfiles', metadata.upload_name),
```

Comment:
`files`, not `virtualfiles`

https://filecache01.rushfiles.com/swagger/ui/index#!/File/File_GetFile

```suggestion
            self._build_filecache_url(str(self.share['id']), 'files', metadata.upload_name),
```